File: spring-boot-validation/src/main/java/com/example/springbootvalidation/exception/ApiError.java
package com.example.springbootvalidation.exception;

import lombok.Data;
import org.springframework.http.HttpStatus;
import org.springframework.validation.FieldError;
import org.springframework.validation.ObjectError;

import java.time.ZonedDateTime;
import java.util.List;

@Data
public class ApiError {

    private HttpStatus status;
    private String clientMessage;
    private List<FieldError> fieldErrors;
    private List<ObjectError> objectErrors;
    private String developerMessage;
    private ZonedDateTime timestamp;
    private String uuid;
    private String path;

    public ApiError(HttpStatus status, String clientMessage, String developerMessage){
        this.status = status;
        this.clientMessage = clientMessage;
        this.developerMessage = developerMessage;
    }

}

File: spring-boot-api-error/src/main/kotlin/com/example/api/errors/ApiError.kt
package com.example.api.errors

import org.springframework.http.HttpStatus

data class ApiError(
    var status: HttpStatus,
    var clientMessage: String,
    var errors: List<String>,
    var developerMessage: String? = null
){
    constructor(status: HttpStatus, clientMessage: String, error: String, developerMessage: String? = null) :
        this(status, clientMessage, arrayListOf<String>(error), developerMessage)
}